packages/utils/crates/utils-types/src/sma.rs

```diff
@@ -1,56 +1,67 @@
 use ringbuffer::{AllocRingBuffer, RingBuffer};
 use std::fmt::{Display, Formatter};
+use std::marker::PhantomData;
 use std::ops::{Add, Div};
 
 /// Simple Moving Average trait.
 /// The second-most useful filter type, bested only by coffee filters.
-pub trait SMA<T>
-where
-    T: for<'a> Add<&'a T, Output = T> + Div<T, Output = T> + Clone + From<u32> + Display + Default,
-{
+pub trait SMA<T> {
     /// Adds a sample.
     fn add_sample(&mut self, sample: T);
```

Comment:
```suggestion
    fn push(&mut self, sample: T);
```

Reply:
The reasoning being - if you do add, it does not imply also "remove if window is full". But push makes no assumptions on side effects.

Reply:
See https://github.com/hoprnet/hoprnet/pull/5744